File: _annexe3 (singleton)/annexe3b/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.widget.SeekBar;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    SeekBar s1;
    SeekBar s2;
    SeekBar s3;

    ArrayList<Integer> liste;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        liste = new ArrayList<>();
        s1 = findViewById(R.id.seekBar);
        s2 = findViewById(R.id.seekBar2);
        s3 = findViewById(R.id.seekBar3);
        s1.setProgress(0);
        s2.setProgress(0);
        s3.setProgress(0);


        try {
            FileInputStream fis = openFileInput("fichier.ser");
            ObjectInputStream ois = new ObjectInputStream(fis);
            liste = (ArrayList<Integer>) ois.readObject();
            s1.setProgress(liste.get(0));
            s2.setProgress(liste.get(1));
            s3.setProgress(liste.get(2));
        } catch (IOException | ClassNotFoundException ignored) {

        }

    }



    @Override
    protected void onStop() {
        super.onStop();

        try {
            FileOutputStream fos =  openFileOutput("fichier.ser", MODE_PRIVATE);
            ObjectOutputStream oos = new ObjectOutputStream(fos);
            liste = new ArrayList<>();
            liste.add(s1.getProgress());
            liste.add(s2.getProgress());
            liste.add(s3.getProgress());
            oos.writeObject(liste);

        } catch (IOException ignored) {

        }


    }
}

File: EXAMEN/_examen1/app/src/main/java/com/example/examen1/MainActivity.kt
package com.example.examen1

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.ObjectInputStream
import java.io.ObjectOutputStream
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import java.util.Scanner
import java.util.Vector


class MainActivity : AppCompatActivity() {
    private lateinit var utilisation: TextView
    private lateinit var langages: TextView
    private var nbLang: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        utilisation = findViewById(R.id.utilisation)
        langages = findViewById(R.id.nb_languages)


        var fis : FileInputStream
        val languagesVector = Vector<Language>()

        try {
            fis = openFileInput("langage.txt")
            fis.use {
                val s = Scanner(fis)
                s.useDelimiter(",")
                s.use {
                    while (s.hasNext()){
                        val nom = s.next()
                        val _2024 = s.next().toInt()
                        val _2019 = s.next().toInt()
                        val _2012 = s.next().toInt()
                        languagesVector.add(Language(nom,_2024, _2019, _2012))
                    }
                }
            }
            for (lang in languagesVector)
                if (lang._2012 == 99)
                    nbLang++
            langages.text = "Nombre de langages qui n'existaient pas en 2012 : " + nbLang.toString()
        }
        catch (e : Exception){
            langages.text = "Fichier de langage non trouvé"
        }




        var temps = ""
        try {
            fis = openFileInput("temps.ser")
            var ois = ObjectInputStream(fis)
            ois.use {
                temps = ois.readObject() as String
            }
            utilisation.text = "Dernière utilisation : " + temps
        }
        catch (e : Exception){
            println("first")
            utilisation.text = "1ère utilisation"
        }


    }

    override fun onStop() {
        super.onStop()


        val date = LocalDateTime.now()
        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
        val text = date.format(formatter)

        val fos : FileOutputStream = openFileOutput("temps.ser", MODE_PRIVATE)
        val oos = ObjectOutputStream(fos)

        oos.writeObject(text)
        oos.close()
        finish();
    }

    internal inner class Language(var nom: String, var _2024: Int, var _2019: Int, var _2012: Int)
}